packages/xod-client/src/core/containers/App.jsx

```diff
@@ -141,28 +138,62 @@ export default class App extends React.Component {
             )(this.props.currentPatchPath),
         })
       )
-    )(eitherTProject);
+      .then(
+        ({ code, nodeIdsMap, nodePinKeysMap, pinsAffectedByErrorRaisers }) =>
+          this.props.actions.runSimulation(
+            explodeMaybe(
+              'currentPatchPath already folded in `transformProjectForTranspiler`',
+              this.props.currentPatchPath
+            ),
+            nodeIdsMap,
+            nodePinKeysMap,
+            code,
+            pinsAffectedByErrorRaisers
+          )
+      )
+      .catch(
+        R.compose(err => {
+          this.props.actions.addError(err);
+          this.props.actions.abortSimulation();
+        }, R.when(R.is(Error), formatErrorMessage))
+      );
   }
 
-  getGlobals() {
-    return catMaybies({
-      XOD_USERNAME: R.compose(R.map(enquote), R.chain(maybeProp('username')))(
-        this.props.user
-      ),
-      XOD_PROJECT: R.compose(
-        R.map(enquote),
-        R.ifElse(R.length, Maybe.of, Maybe.Nothing),
-        getProjectName
-      )(this.props.project),
-    });
+  // :: [LiteralName] -> Promise (Map LiteralName String) Error
+  getGlobals(globalsUsedInProject) {
```

Comment:
```suggestion
  getGlobals(globalNames) {
```

This looks pretty universal regardless are they used or not